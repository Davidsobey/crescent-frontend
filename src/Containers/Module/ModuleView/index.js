/**
 *
 * ModuleView
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { reduxForm } from 'redux-form';
import { CircularProgress } from 'material-ui/Progress';

import Card from '../../../Components/Card';
import Table from '../../../Components/Table';
import CourseActions from '../../../Actions/CourseActions';
import ModuleActions from '../../../Actions/ModuleActions';

const header = ['ID', 'Name', 'Description', 'Category'];

class CourseView extends React.Component {
  constructor(props) {
    super(props);
    this.props.dispatch(CourseActions.getAll());
    this.props.dispatch(ModuleActions.getAll());
  }

  // courses.map((course) => {
  //   if (module.couseID === course.id) {
  //     formattedModule.couseID = course.name;
  //     return formattedModule;
  //   }
  //   return formattedModule;
  // });

  loadData = (modules, courses) => {
    const formattedArray = [];
    if (Array.isArray(modules)) {
      modules.forEach((module) => {
        const courseMatch = courses.filter(course => course.id === module.courseId);
        const newModule = {
          id: module.id,
          description: module.description,
          name: module.name,
          couse: courseMatch[0].name,
        };
        formattedArray.push(newModule);
      });
      formattedArray.sort((a, b) => a.id - b.id);
    }
    return formattedArray;
  };

  render() {
    const { courses, modules } = this.props;
    const data = this.loadData(modules, courses);
    return (
      <Card width="800px" title="Module List">
        {Array.isArray(modules) && Array.isArray(courses) ? (
          <Table header={header} data={data} />
        ) : (
          <div className="center">
            <CircularProgress color="secondary" />
          </div>
        )}
      </Card>
    );
  }
}

const mapStateToProps = state => ({
  courses: state.CourseReducer.courses,
  modules: state.ModuleReducer.modules,
});

const withForm = reduxForm(
  {
    form: 'courseView',
  },
  CourseView,
);

CourseView.propTypes = {
  courses: PropTypes.array,
  dispatch: PropTypes.func,
  modules: PropTypes.array,
};

export default compose(connect(mapStateToProps), withForm)(CourseView);
