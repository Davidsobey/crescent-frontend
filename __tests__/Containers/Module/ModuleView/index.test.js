import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import ModuleView from '../../../../src/Containers/Module/ModuleView/index.js';


const mockStore = configureStore();
const initialState = {
  CourseReducer:{
    courses: [],
    loading: false,
  },
  ModuleReducer:{
    modules: [],
    loading: false,
  }
};
const store = mockStore(initialState);

describe('<ModuleView />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<ModuleView store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});