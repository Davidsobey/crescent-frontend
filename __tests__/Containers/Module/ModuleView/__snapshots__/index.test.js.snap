// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleView /> render() renders the component 1`] = `
<Connect(Form(ModuleView))
  courses={Array []}
  courses_loading={false}
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="courseView"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  modules={Array []}
  modules_loading={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
